src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -924,6 +932,38 @@ bool OutputStateMachineEngine::_ParseHyperlink(const std::wstring_view string,
     return false;
 }
 
+// Routine Description:
+// - OSC 9 ; 4 ; state ; progress ST
+//       state: the state to set the taskbar progress indicator to
+//       progress: the progress value to set the taskbar progress indicator with
+// - Parses out the 'state' and 'progress' parameters from the OSC string so
+//   that we can use them to set the taskbar progress indicator
+// - This follows the ConEmu style, more details here:
+//   https://conemu.github.io/en/AnsiEscapeCodes.html#ConEmu_specific_OSC
+// Arguments:
+// - string: the string to parse
+// - state: where to store the state value once we parse it
+// - progress: where to store the progress value once we parse it
+// Return Value:
+// - true if we successfully parsed the string, false otherwise
+bool OutputStateMachineEngine::_GetTaskbarProgress(const std::wstring_view string,
+                                                   size_t& state,

```

Comment:
This is gonna feel like a nit again, but I think we should probably have the `state` be some sort of named enum value - it can totally stay a `size_t` under the sheets, but I think that would probably be better than just passing a blind `size_t` value around. Check out the other enums in `src\terminal\adapter\DispatchTypes.hpp`

Reply:
TLDR of discussion we had on Teams about this: leave it as is because the `switch` statement based on `state` only happens in `IslandWindow.cpp` and we do not want that to take a dependency on the parser